client/src/lib.rs

```diff
@@ -102,12 +111,19 @@ impl SubxtClient {
     pub fn new(mut task_manager: TaskManager, rpc: RpcHandlers) -> Self {
         let (to_back, from_front) = mpsc::channel(DEFAULT_CHANNEL_SIZE);
 
+        let request_id = Arc::new(RwLock::new(u64::MIN));
```

Comment:
I don't understand why you instantiate, inside `task::spawn` should work?

Reply:
I guess because the main closure is for each `FrontToBack` message. That said the context could be expanded although that is already an extremely indented block of code.